src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb

```diff
@@ -2275,12 +2275,21 @@ End Interface")
 
         <Fact>
         Public Sub TestEnumDeclarationFromSymbol()
-            Dim a As SyntaxNode = Generator.Declaration(_emptyCompilation.GetTypeByMetadataName("System.DateTimeKind"))
             VerifySyntax(Of EnumBlockSyntax)(Generator.Declaration(_emptyCompilation.GetTypeByMetadataName("System.DateTimeKind")),
 "Public Enum DateTimeKind
     Unspecified = 0
     Utc = 1
     Local = 2
+End Enum")
+        End Sub
+
+        <Fact>
+        Public Sub TestEnumWithUnderlyingTypeFromSymbol()
+            VerifySyntax(Of EnumBlockSyntax)(Generator.Declaration(_emptyCompilation.GetTypeByMetadataName("System.Security.SecurityRuleSet")),
+"Public Enum SecurityRuleSet As System.Byte
+    None = CByte(0)
```

Comment:
For some reason in VB it is returning a redundant cast.

Reply:
that's ok.  not an issue for you to deal with. :)

Reply:
Sounds good :)